tokio/src/runtime/basic_scheduler.rs

```diff
@@ -1,22 +1,31 @@
-use crate::park::{Park, Unpark};
+use crate::loom::sync::Mutex;
+use crate::park::{Park, ParkThread, Unpark};
 use crate::runtime;
 use crate::runtime::task::{self, JoinHandle, Schedule, Task};
 use crate::util::linked_list::{Link, LinkedList};
-use crate::util::{waker_ref, Wake};
+use crate::util::{waker_ref, Wake, WakerRef};
 
 use std::cell::RefCell;
 use std::collections::VecDeque;
 use std::fmt;
 use std::future::Future;
-use std::sync::{Arc, Mutex};
-use std::task::Poll::Ready;
+use std::sync::Arc;
+use std::task::{Poll::Ready, Waker};
 use std::time::Duration;
 
 /// Executes tasks on the current thread
-pub(crate) struct BasicScheduler<P>
-where
-    P: Park,
-{
+pub(crate) struct BasicScheduler<P: Park> {
+    /// Inner with the dedicated parker P
+    inner: Mutex<Option<Inner<P>>>,
+
+    /// Queue of threads available to steal the dedicated parker P.
+    steal_queue: Mutex<VecDeque<Waker>>,
```

Comment:
Unless there is a good reason to keep this as two mutexes, i would prefer having a single mutex. Two mutexes are at risk of deadlocks and users must ensure correct ordering is used when acquiring and releasing mutexes.